src/main/java/me/anxuiz/settings/bukkit/plugin/gui/SettingsMenuBuilder.java

```diff
@@ -0,0 +1,130 @@
+import me.anxuiz.settings.Setting;
+import me.anxuiz.settings.SettingManager;
+import me.anxuiz.settings.Toggleable;
+import me.anxuiz.settings.bukkit.PlayerSettings;
+import me.anxuiz.settings.bukkit.plugin.Commands;
+import me.anxuiz.settings.bukkit.plugin.Permissions;
+import me.anxuiz.settings.bukkit.plugin.SettingsCommand;
+import me.anxuiz.settings.bukkit.plugin.gui.clickables.Clickable;
+import org.bukkit.Bukkit;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.Sound;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.inventory.InventoryClickEvent;
+import org.bukkit.event.inventory.InventoryCloseEvent;
+import org.bukkit.inventory.Inventory;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.Set;
+
+public class SettingsMenuBuilder {
+
+    private static final Set<Setting> settings = Sets.newLinkedHashSet(PlayerSettings.getRegistry().getSettings());
+
+    private static final ItemStack BORDER_ICON = new ItemBuilder(Material.STAINED_GLASS_PANE).setDamage((short) 7).setName(ChatColor.GRAY + "#").createItem();
+    private static final boolean BORDER = true;
+
+    private static Inventory makeInventory(boolean border) {
+        int size = settings.size();
+        if (border) {
+            int slots = (int) ((Math.round((size / 7) + 0.5) + 2) * 9);
+            return makeBorder(Bukkit.createInventory(null, slots, "Customize Your Settings!"));
+        }
+        int slots = (int) (Math.round((size / 9) + 0.5) * 9);
+        return Bukkit.createInventory(null, slots, "Customize Your Settings!");
+    }
+
+    /**
+     *                      INVENTORY
+     *
+     * Based on vector i and j notation.
+     * 0i through 8i horizontally
+     * 0j through <code>n</code>j vertically, where <code>n</code> is number of rows
+     *
+     * _______________________________________________________
+     * |  0  |  1  |  2  |  3  |  4  |  5  |  6  |  7  |  8  |
+     * |-----------------------------------------------------|
+     * |  9  | 10  | 11  | 12  | 13  | 14  | 15  | 16  | 17  |
+     * |-----------------------------------------------------|
+     * |  18 | 19  | 20  | 21  | 22  | 23  | 24  | 25  | 26  |
+     * |-----------------------------------------------------|
+     * |  27 | 28  | 29  | 30  | 31  | 32  | 33  | 34  | 35  |
+     * |-----------------------------------------------------|
+     *
+     */
+    private static Inventory makeBorder(Inventory inventory) {
+        int k = 0;
+        for (int j = 1; j <= inventory.getSize() / 9; j++) {
+            for (int i = 1; i <= 9; i++) {
+                // Puts border items in the outer slots. In order of appearance, right column, left column, top row, bottom row
+                if (k == ((9 * j) - 1) || k == ((9 * j) - 9) || k == (i - 1) || k == ((9 * (inventory.getSize() / 9 - 1)) - 1 + i)) {
+                    Clickable click = new Clickable(BORDER_ICON) {
```

Comment:
Every border item has its own click handler, that only cancels clicks to itself, but still receives events from all other items?